bin/io.js

```diff
@@ -0,0 +1,56 @@
+// this module just contains stuff used for io
+const fs = require('fs');
+const util = require('util');
+const httpMessageParser = require('http-message-parser');
+
+/**
+ * This opens a file from disk asynchronously.
+ *
+ * @param {string} path - A file path.
+ *
+ * @returns {string|Buffer} The result of opening the file.
+ */
+const readFile = util.promisify(fs.readFile);
+exports.readFile = readFile;
+
+/**
+ * Simple wrapper around node's process.stdin.
+ * This allows us to get the result of cat and other commands.
+ *
+ * @param {string} [encoding='utf8'] - The http message's encoding.
+ *
+ * @returns {Promise<string>} Should return the http message.
+ */
+async function getStdin(encoding = 'utf8') {
```

Comment:
How do you feel about this syntax?
```
exports.foo = async () => {
```

Reply:
not that hot on it.
If I don't need an arrow function i.e. I'm not inside an es6 class or an existing function then I think it's better to just use vanilla function statements. also using variable assignment on an arrow function means the function is not defined in javascript's parse step. function statements means the debugger knows the name of the functions your arrow function is technically speaking nameless. It's not a huge deal either way, but I do prefer function statements over unnamed lambdas unless defining an inline helper in say .map or .reduce etc.

Reply:
> function statements means the debugger knows the name of the functions your arrow function is technically speaking nameless.

Just want to point out that we tested this -- the debugger knows either way.

Reply:
https://stackoverflow.com/a/48764221/399274

Reply:
I had a feeling I would be wrong, but I still think function statements are better than assigned lambdas. so I'm going to skip this one.